### DISCUSSION TITLE
Power Flow: Quantum Fast Decoupled Load Flow

### DISCUSSION THREAD

**Discussion by TimThacker** [CURENT/andes#391] [Ideas]
Hantao,

I am considering using IBM's [qiskit](https://qiskit.org/textbook/ch-applications/hhl_tutorial.html) implementation of the HHL algorithm to solve a system of linear equations using a simulated quantum circuit. Ideally, I would use this in the fast decoupled load flow method to create a new type of power flow. Should I have any concerns about ANDES compatibility with qiskit?

**Comment by cuihantao** [comment]
I haven't looked into qiskit and not sure to which degree you will need to reimplement the data structures, program structure, and algorithms. I'm not sure if ANDES is compatible with qiskit. Although they are both in Python, the data structures qiskit support and the ways in which algorithms need to be written may be vastly different than ANDES.

**Reply by TimThacker** [reply]
Hantao,

I have spent several hours looking through this. I believe it is possible, but I need some help getting the proper submatrices of the Jacobian for fast decoupled load flow. I have been looking through the way ANDES handles the data, and it appears that the equations are grouped into differential and algebraic equations. But I do not know how to know which indices correspond to which power flow equations.  Could you help me with this?

**Reply by cuihantao** [reply]
Sure. That is a major part of the internal program so you will need to start by reading the code. Check `System.set_address` for the variable index assignment. For power flow, you will mostly only have algebraic equations. Check the equations associated with variables in power flow models, see Bus, PV, PQ, Slack, and Shunt.
